hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/Downloader.java

```diff
@@ -532,4 +536,36 @@ private Counter newStatusMetric(Long nodeId, String streamType, SignatureStatus
                 .tag("status", status.toString())
                 .register(meterRegistry);
     }
+
+    /**
+     * Returns a randomly-selected (and randomly-ordered) collection of CondensusNode elements from the
+     * input, where the total stake is just enough to meet/exceed the CommonDownloader "downloadRatio" property.
+     *
+     * @param allNodes the entire set of ConsensusNodes
+     * @return a randomly-ordered subcollection
+     */
+    @VisibleForTesting
+    Collection<ConsensusNode> partialCollection(Collection<ConsensusNode> allNodes) {
+        var nodes = new ArrayList<ConsensusNode>(allNodes);
+        var downloadRatio = downloaderProperties.getCommon().getDownloadRatio();
+        // no need to randomize (just return entire list) if # of nodes is 0 or 1 or downloadRatio == 1
+        if (nodes.size() <= 1 || downloadRatio.compareTo(BigDecimal.ONE) == 0) {
+            log.debug("partialCollection: Kept all {} nodes", nodes.size());
+            return nodes;
+        }
+        // shuffle nodes into a random order
+        Collections.shuffle(nodes);
+
+        // only keep "just enough" nodes to reach/exceed downloadRatio
+        BigDecimal neededStake = BigDecimal.valueOf(nodes.get(0).getTotalStake()).multiply(downloadRatio);
```

Comment:
I think we should get the long value of `neededStake` with `RoundingMode.CEILING` so we can avoid aggregating `BigDecimal` aggreatedStake

 ```suggestion
        long neededStake = BigDecimal.valueOf(nodes.get(0).getTotalStake())
          .multiply(downloadRatio)
          .setScale(0, RoundingMode.CEILING)
          .longValue();
```

Reply:
Done.  Sounds good!